packages/rooks/src/hooks/useArrayState.ts

```diff
@@ -0,0 +1,90 @@
+import { UseArrayStateControls, UseArrayStateReturnValue } from "@/types/types";
+import { useCallback, useMemo, useState } from "react";
+
+/**
+ * useArrayState
+ * @description Array state manager hook for React
+ * @param {Array<T>} initialState Initial state of the array
+ * @returns {UseArrayStateReturnValue<T>} Array state manager hook for React
+ * @see {@link https://react-hooks.org/docs/useArrayState}
+ *
+ * @example
+ *
+ * const [array, controls] = useArrayState([1, 2, 3]);
+ *
+ * controls.push(4); // [1, 2, 3, 4]
+ * controls.pop(); // [1, 2, 3]
+ * controls.unshift(0); // [0, 1, 2, 3]
+ * controls.shift(); // [1, 2, 3]
+ * controls.reverse(); // [3, 2, 1]
+ * controls.concat([4, 5, 6]); // [1, 2, 3, 4, 5, 6]
+ * controls.fill(0); // [0, 0, 0, 0, 0, 0]
+ */
+function useArrayState<T>(initialArray: T[] = []): UseArrayStateReturnValue<T> {
+  const [array, setArray] = useState(initialArray);
+
+  const push = useCallback(
+    (value: T) => {
```

Comment:
The arguments are not following the same pattern as the type definitions: `args...: Parameters<Array<T>["push"]>` etc.